smokey/src/main/java/frc/robot/subsystems/DrivetrainSubsystem.java

```diff
@@ -616,6 +615,25 @@ private ChassisSpeeds clampChassisSpeeds(ChassisSpeeds chassisSpeeds){
     MAX_ANGULAR_VELOCITY_RADIANS_PER_SECOND);
   }
 
+  private ChassisSpeeds discretize(ChassisSpeeds speeds) {
+    // a fudge factor to increase the size of the discretization correction. 
+    // other teams use [1..4]
+    double timeScaleFactor = 1.9; 
+    var desiredDeltaPose = new Pose2d(
+      speeds.vxMetersPerSecond * deltaTimeSeconds, 
+      speeds.vyMetersPerSecond * deltaTimeSeconds, 
+      new Rotation2d(speeds.omegaRadiansPerSecond * deltaTimeSeconds * timeScaleFactor)
+    );
+    var twist = new Pose2d().log(desiredDeltaPose);
+
+    return new ChassisSpeeds(
+      (twist.dx / deltaTimeSeconds), 
```

Comment:
I also read ... https://first.wpi.edu/wpilib/allwpilib/docs/release/java/edu/wpi/first/math/geometry/Twist2d.html 

![image](https://github.com/Team4682CyBears/ChargedUp2023/assets/56608471/31f95096-d50c-43aa-962d-690239d9ad4c)

Reply:
seems this turns it back into a velocity ... is that correct?

Reply:
Yep.